backend/endpoints/sockets/scan.py

```diff
@@ -135,176 +154,221 @@ async def stop_scan():
+        return
+    except Exception as e:
+        log.error(e)
+        # Catch all exceptions and emit error to the client
+        await sm.emit("scan:done_ko", str(e))
+        return
 
-            platform = db_platform_handler.add_platform(scanned_platform)
 
-            await sm.emit(
-                "scan:scanning_platform",
-                PlatformSchema.model_validate(platform).model_dump(
-                    include={"id", "name", "slug"}
-                ),
-            )
-            await sm.emit("", None)
+async def _identify_platform(
+    platform_slug: str,
+    scan_type: ScanType,
+    fs_platforms: list[str],
+    selected_roms: list[str],
+    metadata_sources: list[str],
+    socket_manager: socketio.AsyncRedisManager,
+) -> ScanStats:
+    # Stop the scan if the flag is set
+    if redis_client.get(STOP_SCAN_FLAG):
+        raise ScanStoppedException()
 
-            # Scanning firmware
-            try:
-                fs_firmware = fs_firmware_handler.get_firmware(platform)
-            except FirmwareNotFoundException:
-                fs_firmware = []
+    scan_stats = ScanStats()
 
-            if len(fs_firmware) == 0:
-                log.warning(
-                    "  ⚠️ No firmware found, skipping firmware scan for this platform"
-                )
-            else:
-                log.info(f"  {len(fs_firmware)} firmware files found")
+    platform = db_platform_handler.get_platform_by_fs_slug(platform_slug)
+    if platform and scan_type == ScanType.NEW_PLATFORMS:
+        return scan_stats
 
-            for fs_fw in fs_firmware:
-                # Break early if the flag is set
-                if redis_client.get(STOP_SCAN_FLAG):
-                    break
+    scanned_platform = scan_platform(
+        platform_slug, fs_platforms, metadata_sources=metadata_sources
+    )
+    if platform:
+        scanned_platform.id = platform.id
+        # Keep the existing ids if they exist on the platform
+        scanned_platform.igdb_id = scanned_platform.igdb_id or platform.igdb_id
+        scanned_platform.moby_id = scanned_platform.moby_id or platform.moby_id
+
+    scan_stats.scanned_platforms += 1
+    scan_stats.added_platforms += 1 if not platform else 0
+    scan_stats.metadata_platforms += (
+        1 if scanned_platform.igdb_id or scanned_platform.moby_id else 0
+    )
 
-                firmware = db_firmware_handler.get_firmware_by_filename(
-                    platform.id, fs_fw
-                )
+    platform = db_platform_handler.add_platform(scanned_platform)
 
-                scanned_firmware = scan_firmware(
-                    platform=platform,
-                    file_name=fs_fw,
-                    firmware=firmware,
-                )
+    await socket_manager.emit(
+        "scan:scanning_platform",
+        PlatformSchema.model_validate(platform).model_dump(
+            include={"id", "name", "slug"}
+        ),
+    )
+    await socket_manager.emit("", None)
```

Comment:
what's this empty one for?

Reply:
I'm not sure what we use these messages for, but this is ported as-is (line 170 in current `master`).

Reply:
This is to flush the buffer so the messages are sent on real time to the client